### PR TITLE
Feature/docs 2175 migrate to all in one deployment

### PR DESCRIPTION
**Pull request by MaximBashurov**

### CODE REVIEW

File: docs/latest/updating-migrating/older-versions/nginx-modules.md [hunk 1]
```diff
@@ -22,36 +22,276 @@
...
-## Requirements
+## Inform Wallarm technical support that you are upgrading EOL  node
+
+If upgrading the end‑of‑life Wallarm NGINX modules (version 3.6 and lower) to version 4.6, inform [Wallarm technical support](mailto:support@wallarm.com) about that and ask for assistance.
+
+Besides any other help, ask to enable new IP lists logic for your Wallarm account. When new IP lists logic is enabled, please open Wallarm Console and ensure that the section [**IP lists**](../../user-guides/ip-lists/overview.md) is available.
+
+## Upgrade methods
+
+You can upgrade the the end‑of‑life Wallarm NGINX modules (version 3.6 and lower) installed from individual DEB/RPM packages to version 4.6 in two different ways:
```
Comment (AnastasiaTWW) [inline docs/latest/updating-migrating/older-versions/nginx-modules.md:39]: double the

File: docs/latest/updating-migrating/older-versions/nginx-modules.md [hunk 2]
```diff
@@ -22,36 +22,276 @@
...
+
+* Migrate to the [all-in-one installer](#upgrade-with-all-in-one-installer) usage during the upgrade procedure. This is the recommended approach as it automates various node installation and upgrade activities, such as NGINX and OS version identification, adding appropriate Wallarm repositories and installing packages, and others.
+* Keep using the current [manual](#manual-upgrade) installation method If you prefer to stick with the current installation method using individual DEB/RPM packages. However, it's important to note that this approach might require additional effort and manual configuration during the upgrade process in comparison to the new method for Wallarm node installation on Debian/Ubuntu operating systems.
```
Comment (AnastasiaTWW) [inline docs/latest/updating-migrating/older-versions/nginx-modules.md:42]: check. for some reasons "If you prefer" starts with a capital letter

File: docs/latest/updating-migrating/older-versions/nginx-modules.md [hunk 3]
```diff
@@ -22,36 +22,276 @@
```
Comment (AnastasiaTWW) [inline docs/latest/updating-migrating/older-versions/nginx-modules.md:22]: mkdocs serve returns errors

File: include/waf/installation/all-in-one-upgrade-requirements.md [hunk 4]
```diff
@@ -0,0 +1,5 @@
+* Access to the account with the **Administrator** role in Wallarm Console for the [US Cloud](https://us1.my.wallarm.com/) or [EU Cloud](https://my.wallarm.com/).
+* Access to `https://meganode.wallarm.com` to download all-in-one Wallarm installer. Ensure the access is not blocked by a firewall.
+* Access to `https://us1.api.wallarm.com` for working with US Wallarm Cloud or to `https://api.wallarm.com` for working with EU Wallarm Cloud. If access can be configured only via the proxy server, then use the [instructions][configure-proxy-balancer-instr].
+* Executing all commands as a superuser (e.g. `root`).
+* Using [new clean machine](#step-1-prepare-clean-machine) for node installation.
```
Comment (AnastasiaTWW) [inline include/waf/installation/all-in-one-upgrade-requirements.md:5]: why is it a requirement? it is a step

File: docs/latest/updating-migrating/older-versions/nginx-modules.md [hunk 5]
```diff
@@ -22,36 +22,276 @@
...
+--8<-- "../include/waf/installation/all-in-one-upgrade-requirements.md"
+
+### Upgrade procedure
+
+* If filtering node and postanalytics modules are installed on the same server, then follow the instructions below to upgrade all.
+
+    You will need to run a node of the newer version using all-in-one installer on a clean machine, test that it works well and stop the previous one and configure traffic to flow through the new machine instead of the previous one.
+
+* If filtering node and postanalytics modules are installed on different servers, **first** upgrade the postanalytics module and **then** the filtering module following these [instructions](separate-postanalytics.md).
+
```
Comment (AnastasiaTWW) [inline docs/latest/updating-migrating/older-versions/nginx-modules.md:59]: where are the steps for active threat bverification enabling/disabling?

File: docs/latest/updating-migrating/older-versions/nginx-modules.md [hunk 6]
```diff
@@ -22,36 +22,276 @@
...
+Using new clean machine will lead to that at some moment you will have both old and new node, which is good: you can test the new one working properly without stopping the old one.
+
+### Step 2: Install NGINX and dependencies
+
+--8<-- "../include/waf/installation/all-in-one-nginx.md"
+
+### Step 3: Prepare Wallarm token
+
+--8<-- "../include/waf/installation/all-in-one-token.md"
+
```
Comment (AnastasiaTWW) [inline docs/latest/updating-migrating/older-versions/nginx-modules.md:82]: layout is broken in several places - open the built docs and search for "]" on a pge

File: docs/latest/updating-migrating/older-versions/nginx-modules.md [hunk 7]
```diff
@@ -22,36 +22,276 @@
...
+1. Send the request with test [SQLI][sqli-attack-docs] and [XSS][xss-attack-docs] attacks to the protected resource address:
+
+    ```
+    curl http://localhost/?id='or+1=1--a-<script>prompt(1)</script>'
+    ```
+
+1. Open the Wallarm Console → **Events** section in the [US Cloud](https://us1.my.wallarm.com/search) or [EU Cloud](https://my.wallarm.com/search) and ensure attacks are displayed in the list.
+1. As soon as your Cloud stored data (rules, IP lists) is synchronized to the new node, perform some test attacks to make sure your rules work as expected.
+
+### Step 10: Update the Wallarm blocking page
```
Comment (AnastasiaTWW) [inline docs/latest/updating-migrating/older-versions/nginx-modules.md:227]: such setting must happen before testing

File: docs/latest/updating-migrating/older-versions/nginx-modules.md [hunk 8]
```diff
@@ -22,36 +22,276 @@
...
+
+=== "In-line"
+    Update targets of your load balancer to send traffic to the Wallarm instance. For details, please refer to the documentation on your load balancer.
+
+    Before full redirecting of the traffic to the new node, it is recommended to first redirect it partially and check that the new node behaves as expected.
+
+=== "Out-of-Band"
+    Configure your web or proxy server (e.g. NGINX, Envoy) to mirror incoming traffic to the Wallarm node. For configuration details, we recommend to refer to your web or proxy server documentation.
+
+    Inside the [link][web-server-mirroring-examples], you will find the example configuration for the most popular of web and proxy servers (NGINX, Traefik, Envoy).
```
Comment (AnastasiaTWW) [inline docs/latest/updating-migrating/older-versions/nginx-modules.md:245]: broken layout again

File: docs/latest/updating-migrating/older-versions/nginx-modules.md (hunk 5 above)
Comment (AnastasiaTWW) [reply docs/latest/updating-migrating/older-versions/nginx-modules.md:59]: the same for api port

File: docs/latest/updating-migrating/older-versions/separate-postanalytics.md [hunk 9]
```diff
@@ -1,23 +1,95 @@
...
+
 
 # Upgrading the EOL postanalytics module
 
 These instructions describe the steps to upgrade the end‑of‑life postanalytics module (version 3.6 and lower) installed on a separate server. Postanalytics module must be upgraded before [Upgrading Wallarm NGINX modules][docs-module-update].
 
 --8<-- "../include/waf/upgrade/warning-deprecated-version-upgrade-instructions.md"
 
-## Requirements
+## Inform Wallarm technical support that you are upgrading EOL node
```
Comment (AnastasiaTWW) [inline docs/latest/updating-migrating/older-versions/separate-postanalytics.md:17]: no need for that here

File: docs/latest/updating-migrating/older-versions/nginx-modules.md (hunk 5 above)
Comment (MaximBashurov) [reply docs/latest/updating-migrating/older-versions/nginx-modules.md:59]: Comments #1-2 are fixed. In the separate postanalytics guide, the API port info is added, but the active threat verification steps were never presented (they are absent in the current "manual" approach).

File: docs/latest/updating-migrating/older-versions/separate-postanalytics.md (hunk 9 above)
Comment (MaximBashurov) [reply docs/latest/updating-migrating/older-versions/separate-postanalytics.md:17]: Why not?

File: docs/latest/updating-migrating/older-versions/separate-postanalytics.md [hunk 10]
```diff
@@ -1,23 +1,95 @@
...
+
+Use the procedure below to upgrade the end‑of‑life postanalytics module (version 3.6 and lower) installed on a separate server to version 4.6 using [all-in-one installer](../../installation/nginx/all-in-one.md).
+
+### Requirements for upgrade using all-in-one installer
+
+--8<-- "../include/waf/installation/all-in-one-upgrade-requirements.md"
+
+### Step 1: Prepare clean machine
+
+When upgrading from the end‑of‑life postanalytics module (version 3.6 and lower) to 4.6 with all-in-one installer, you cannot upgrade an old package installation - instead you need to use a clean machine. Thus, as step 1, prepare a machine with one of the supported OS:
```
[the remark this answers is not in the document]
Comment (MaximBashurov) [reply docs/latest/updating-migrating/older-versions/separate-postanalytics.md:38]: The introduction sentence is different in all 4 cases - I do not think we need to put the remaining part in include.

File: docs/latest/updating-migrating/nginx-modules.md [hunk 11]
```diff
@@ -37,8 +37,8 @@ To upgrade the end‑of‑life node (3.6 or lower), please use the [different in
 
 You can upgrade the Wallarm NGINX modules 4.x installed from individual DEB/RPM packages to version 4.6 in two different ways:
 
-* Migrate to the [all-in-one installer](#upgrade-with-all-in-one-installer) usage during the upgrade procedure. This is the recommended approach as it automates various node installation and upgrade activities, such as NGINX and OS version identification, adding appropriate Wallarm repositories and installing packages, and others.
-* Keep using the current [manual](#manual-upgrade) installation method If you prefer to stick with the current installation method using individual DEB/RPM packages. However, it's important to note that this approach might require additional effort and manual configuration during the upgrade process in comparison to the new method for Wallarm node installation on Debian/Ubuntu operating systems.
```
Comment (AnastasiaTWW) [inline docs/latest/updating-migrating/nginx-modules.md:41]: why did you delete all details?

File: docs/latest/updating-migrating/older-versions/separate-postanalytics.md [hunk 12]
```diff
@@ -1,23 +1,95 @@
...
+## Inform Wallarm technical support that you are upgrading EOL node
+
+If upgrading the end‑of‑life postanalytics module (version 3.6 and lower) to version 4.6, inform [Wallarm technical support](mailto:support@wallarm.com) about that and ask for assistance.
+
+## Upgrade methods
+
+You can upgrade the end‑of‑life postanalytics module (version 3.6 and lower) installed on a separate server to version 4.6 in two different ways:
+
+* Migrate to the [all-in-one installer](#upgrade-with-all-in-one-installer) usage during the upgrade procedure. This is the recommended approach as it automates various postanalytics module installation and upgrade activities, such as OS version identification, adding appropriate Wallarm repositories and installing packages, and others.
+* Keep using the current [manual](#manual-upgrade) installation method. However, it's important to note that this approach might require additional effort and manual configuration during the upgrade process in comparison to the new all-in-one method.
```
[the remark this answers is not in the document]
Comment (AnastasiaTWW) [reply docs/latest/updating-migrating/older-versions/separate-postanalytics.md]: probably this was unclear. in the current upgrade procedure, nothing is simplified. but in further node upgrade, everything will be easier. you have deleted important info but it is needed just to make it more accurate

File: docs/latest/updating-migrating/older-versions/nginx-modules.md [hunk 13]
```diff
@@ -207,11 +217,27 @@ server {
 
 [More details on the statistics service configuration](../../admin-en/configure-statistics-service.md)
 
-### Step 8: Restart NGINX
+#### Update API port
```
Comment (AnastasiaTWW) [inline docs/latest/updating-migrating/older-versions/nginx-modules.md:220]: it is not about nginx configuration

File: docs/latest/updating-migrating/older-versions/nginx-modules.md [hunk 14]
```diff
@@ -207,11 +217,27 @@ server {
...
+Learn the [recommendation on the Active threat verification module setup](../../admin-en/attack-rechecker-best-practices.md) and re-enable it if required.
+
+After a while, ensure the module operation does not cause false positives. If discovering false positives, please contact the [Wallarm technical support](mailto:support@wallarm.com).
+
+### Step 10: Restart NGINX
 
 --8<-- "../include/waf/installation/restart-nginx-systemctl.md"
 
-### Step 9: Test Wallarm node operation
+### Step 11: Update the Wallarm blocking page
```
Comment (AnastasiaTWW) [inline docs/latest/updating-migrating/older-versions/nginx-modules.md:234]: it is about nginx configuratuon

File: docs/latest/updating-migrating/older-versions/separate-postanalytics.md [hunk 15]
```diff
@@ -1,23 +1,100 @@
...
+
 
 # Upgrading the EOL postanalytics module
 
 These instructions describe the steps to upgrade the end‑of‑life postanalytics module (version 3.6 and lower) installed on a separate server. Postanalytics module must be upgraded before [Upgrading Wallarm NGINX modules][docs-module-update].
 
 --8<-- "../include/waf/upgrade/warning-deprecated-version-upgrade-instructions.md"
 
-## Requirements
+## Inform Wallarm technical support that you are upgrading EOL postanalytics module
```
Comment (AnastasiaTWW) [inline docs/latest/updating-migrating/older-versions/separate-postanalytics.md:18]: we have discussed that this should be deleted

File: docs/latest/updating-migrating/older-versions/separate-postanalytics.md (hunk 10 above)
[the remark this answers is not in the document]
Comment (AnastasiaTWW) [reply docs/latest/updating-migrating/older-versions/separate-postanalytics.md:39]: but why is it different? can it be the same? without specific version numbers

File: docs/latest/updating-migrating/all-in-one.md [hunk 16]
```diff
@@ -0,0 +1,109 @@
+[statistics-service-all-parameters]:        ../admin-en/configure-statistics-service.md
+[img-attacks-in-interface]:                 ../images/admin-guides/test-attacks-quickstart.png
+[tarantool-status]:                         ../images/tarantool-status.png
+
+# Upgrading Wallarm node with All-in-One Installer
```
Comment (AnastasiaTWW) [inline docs/latest/updating-migrating/all-in-one.md:5]: the name for this article and for another nginx module upgrade must differ in the left menu considering specificities of each method

Comment (AnastasiaTWW) [reply docs/latest/updating-migrating/all-in-one.md:5]: Upgrading Node Installed via the All-in-One Installer - is not it more correct?

File: docs/latest/updating-migrating/all-in-one.md [hunk 17]
```diff
@@ -0,0 +1,109 @@
...
+
+These instructions describe the steps to upgrade the Wallarm node 4.6.x installed using [all-in-one installer](../installation/nginx/all-in-one.md) to version 4.6.x+.
+
+## Requirements for upgrade using all-in-one installer
+
+--8<-- "../include/waf/installation/all-in-one-upgrade-requirements.md"
+
+## Prepare Wallarm token
+
+To upgrade node, you will need a Wallarm token of [one of the types](../user-guides/nodes/nodes.md#api-and-node-tokens-for-node-creation). To prepare a token:
```
Comment (AnastasiaTWW) [inline docs/latest/updating-migrating/all-in-one.md:15]: is it the token that was previously generated (during installation)? and this is the step of upgrade procedure - why not to move to the corresponding section?

File: docs/latest/updating-migrating/all-in-one.md [hunk 18]
```diff
@@ -0,0 +1,109 @@
...
+    For upgrade, use the same node token that was used for installation:
+
+    1. Open Wallarm Console → **Nodes** in the [US Cloud](https://us1.my.wallarm.com/nodes) or [EU Cloud](https://my.wallarm.com/nodes).
+    1. In your existing node group, copy token using node's menu → **Copy token**.
+
+## Upgrade procedure
+
+The upgrade procedure differs depending on how filtering node and postanalytics modules are installed:
+
+*  [On the same server](#filtering-node-and-postanalytics-on-the-same-server): modules are upgraded altogether
```
Comment (AnastasiaTWW) [inline docs/latest/updating-migrating/all-in-one.md:34]: double indents

File: docs/latest/updating-migrating/all-in-one.md [hunk 19]
```diff
@@ -0,0 +1,109 @@
...
+
+## Filtering node and postanalytics on the same server
+
+Use the procedure below to upgrade altogether the filtering node and postanalytics modules installed using all-in-one installer on the same server.
+
+### Step 1: Download newest version of all-in-one Wallarm installer
+
+--8<-- "../include/waf/installation/all-in-one-installer-download.md"
+
+### Step 2: Run all-in-one Wallarm installer
```
Comment (AnastasiaTWW) [inline docs/latest/updating-migrating/all-in-one.md:45]: did you test this flow?

File: docs/latest/updating-migrating/all-in-one.md [hunk 20]
```diff
@@ -0,0 +1,109 @@
...
+    curl http://localhost/?id='or+1=1--a-<script>prompt(1)</script>'
+    ```
+
+1. Open the Wallarm Console → **Events** section in the [US Cloud](https://us1.my.wallarm.com/search) or [EU Cloud](https://my.wallarm.com/search) and ensure attacks are displayed in the list.
+1. As soon as your Cloud stored data (rules, IP lists) is synchronized to the new node, perform some test attacks to make sure your rules work as expected.
+
+## Filtering node and postanalytics on different servers
+
+!!! warning "Sequence of steps to upgrade the filtering node and postanalytics modules"
+    If the filtering node and postanalytics modules are installed on different servers, then it is required to upgrade the postanalytics packages before updating the filtering node packages.
```
Comment (AnastasiaTWW) [inline docs/latest/updating-migrating/all-in-one.md:69]: where is the link to separate postanalytics upgarde?

File: docs/latest/updating-migrating/older-versions/separate-postanalytics.md (hunk 9 above)
Comment (MaximBashurov) [reply docs/latest/updating-migrating/older-versions/separate-postanalytics.md]: Discussed, deleted.

File: docs/latest/updating-migrating/all-in-one.md (hunk 17 above)
Comment (MaximBashurov) [reply docs/latest/updating-migrating/all-in-one.md:15]: > is it the token that was previously generated (during installation)

Yes, if it is a node token. And can be any if it is an API token. The section is here as it relates to both major sections (on one server, on different servers).

File: docs/latest/updating-migrating/all-in-one.md (hunk 20 above)
Comment (MaximBashurov) [reply docs/latest/updating-migrating/all-in-one.md:69]: As we discussed, this IS the section that includes the steps for a separate  postanalytics.

File: mkdocs.yml [hunk 21]
```diff
@@ -411,6 +411,7 @@ nav:
       - Filtering node upgrade instructions:
         - Upgrading Wallarm NGINX modules: updating-migrating/nginx-modules.md
```
Comment (AnastasiaTWW) [inline mkdocs.yml:412]: this name does not say anything about difference with Upgrading Node Installed via All-in-One Installer. mention individual packages here

File: docs/latest/updating-migrating/older-versions/separate-postanalytics.md (hunk 12 above)
[the remark this answers is not in the document]
Comment (AnastasiaTWW) [reply docs/latest/updating-migrating/older-versions/separate-postanalytics.md]: not fixed

File: docs/latest/updating-migrating/older-versions/separate-postanalytics.md (hunk 10 above)
[the remark this answers is not in the document]
Comment (AnastasiaTWW) [reply docs/latest/updating-migrating/older-versions/separate-postanalytics.md:32]: not fixed

File: docs/latest/updating-migrating/older-versions/separate-postanalytics.md (hunk 12 above)
[the remark this answers is not in the document]
Comment (MaximBashurov) [reply docs/latest/updating-migrating/older-versions/separate-postanalytics.md]: Discussed, this is fixed.

File: docs/latest/updating-migrating/all-in-one.md (hunk 17 above)
Comment (AnastasiaTWW) [reply docs/latest/updating-migrating/all-in-one.md]: still, this is the step. some steps can be repeated in both sections, that is okay. otherwise, users can miss that. in addition to that, for the postanalytics and NGINX-Wallarm modules there can be cases when users use several tokens

File: docs/latest/updating-migrating/all-in-one.md [hunk 22]
```diff
@@ -0,0 +1,117 @@
+[statistics-service-all-parameters]:        ../admin-en/configure-statistics-service.md
+[img-attacks-in-interface]:                 ../images/admin-guides/test-attacks-quickstart.png
+[tarantool-status]:                         ../images/tarantool-status.png
+
```
Comment (AnastasiaTWW) [inline docs/latest/updating-migrating/all-in-one.md:4]: maaaany places where layout is broken

File: docs/latest/updating-migrating/all-in-one.md [hunk 23]
```diff
@@ -0,0 +1,117 @@
...
+# Upgrading Wallarm node with All-in-One Installer
+
+These instructions describe the steps to upgrade the Wallarm node 4.6.x installed using [all-in-one installer](../installation/nginx/all-in-one.md) to version 4.6.x+.
+
+## Requirements for upgrade using all-in-one installer
```
Comment (AnastasiaTWW) [inline docs/latest/updating-migrating/all-in-one.md:9]: "Requirements" is enough

File: docs/latest/updating-migrating/separate-postanalytics.md [hunk 24]
```diff
@@ -1,47 +1,32 @@
 [docs-module-update]:           nginx-modules.md
 [img-wl-console-users]:         ../images/check-users.png 
 [img-create-wallarm-node]:      ../images/user-guides/nodes/create-cloud-node.png
+[img-attacks-in-interface]:     ../images/admin-guides/test-attacks-quickstart.png
 [wallarm-token-types]:          ../user-guides/nodes/nodes.md#api-and-node-tokens-for-node-creation
 [tarantool-status]:             ../images/tarantool-status.png
+[statistics-service-all-parameters]: ../admin-en/configure-statistics-service.md
```
Comment (AnastasiaTWW) [inline docs/latest/updating-migrating/separate-postanalytics.md:8]: broken layout in this article as well

File: docs/latest/updating-migrating/all-in-one.md (hunk 17 above)
Comment (MaximBashurov) [reply docs/latest/updating-migrating/all-in-one.md]: Fixed

File: include/waf/installation/all-in-one-clean-machine.md [hunk 25]
```diff
@@ -0,0 +1,12 @@
+When upgrading modules 4.x or lower to 4.6 with all-in-one installer, you cannot upgrade an old package installation - instead you need to use a clean machine. Thus, as step 1, prepare a machine with one of the supported OS:
```
[the remark this answers is not in the document]
Comment (MaximBashurov) [reply include/waf/installation/all-in-one-clean-machine.md:1]: Yes, that is why it says "upgrading modules 4.x or lower".
